State.cpp

```diff
@@ -17,27 +21,17 @@ ExpStateVC* ExpStateVC::instance = 0;
 PeepStateVC* PeepStateVC::instance = 0;
 ExpHoldStateVC* ExpHoldStateVC::instance = 0;
 
-BreathData* State::breath;
-
 /*********************************************************/
 /* OffState                                              */
 /*********************************************************/
 
-// performs actions necessary upon entering a state
-static State* OffState::begin(BreathData b) {
-    if (instance==0) instance = new OffState();  // one-time setup
-    breath = &b;                                 // store reference to breath data
-
-    // valve settings
-    inspValve.endBreath(); // close the inspiratory valve
-    expValve.open();       // keep expiratory valve open for safety (also does not use as much power)
-
-    return instance;
-}
-
 static State* OffState::enter() {
-    // if we enter this state other than at the beginning, sound an alarm        
-    alarmMgr.activateAlarm(ALARM_SHUTDOWN);
+    if (instance==0) {
+        instance = new OffState();  // one-time setup
```

Comment:
If you're calling `new`, you're doing something wrong.  This is a classic singleton. There is no reason its space cannot be allocated statically before the program even begins running.  See my comments about globals and singletons, above

Let the compiler do the work.  There is nowhere in this entire project where heap allocation is needed (except maybe for strings in the display code, but even then, I doubt it).  Note that I still believe that `enter()` should be a virtual function (or at least a non-static class function).